further_link/util/upload.py

```diff
@@ -32,6 +34,25 @@ def directory_is_valid(directory):
     )
 
 
+def create_directory(directory_path: str, user: str = None):
+    """
+    Create the directories from the provided path level by level, making sure the folders have the correct owner
+    """
+    if user is None:
+        user = get_user_using_first_display()
```

Comment:
in other places i'm using `utils.user_config.default_user`, which basically uses hardcoded 'pi' user if it's available. this is better, so I think we should change all those places to use `utils.sdk.get_user_using_first_display` for consistency please